src/util/queryNodeMatchers.ts

```diff
@@ -0,0 +1,72 @@
+import { Language, QueryCapture, SyntaxNode, Query } from "web-tree-sitter";
+import {
+  NodeMatcher,
+  SelectionExtractor,
+  SelectionWithEditor,
+} from "../typings/Types";
+import { simpleSelectionExtractor } from "./nodeSelectors";
+
+function getQuery(node: SyntaxNode, scopeQuery: string): Query {
+  const language = node.tree.getLanguage() as Language;
+  return language.query(scopeQuery);
+}
+
+export function defaultMatcher(
+  nodeNames: string[] | string,
+  scopeQuery: string,
+  selector: SelectionExtractor = simpleSelectionExtractor
+): NodeMatcher {
+  return (selection: SelectionWithEditor, node: SyntaxNode) => {
+    let pred = Array.isArray(nodeNames) ? nodeNames : [nodeNames];
+    const query = getQuery(node, scopeQuery);
```

Comment:
This is really the result of running the query, no?
```suggestion
    const queryResult = runQuery(node, scopeQuery);
```

Reply:
We have it when invoking `defaultMatcher`. I'll ensure its pre-compiled.